src/rust/ensogl/lib/core/src/display/scene/layer.rs

```diff
@@ -179,23 +137,52 @@ impl Deref for Layer {
     }
 }
 
+impl Debug for Layer {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        Debug::fmt(&*self.model,f)
+    }
+}
+
 impl Layer {
-    fn new
-    (logger:&Logger, id:LayerId, reg:&Rc<RefCell<LayersRegistry>>, on_mut:Box<dyn Fn()>) -> Self {
-        let model = LayerModel::new(logger,id,reg,on_mut);
+    /// Constructor.
+    pub fn new(logger:Logger) -> Self {
+        let model = LayerModel::new(logger);
         let model = Rc::new(model);
         Self {model}
     }
 
-    fn downgrade(&self) -> WeakLayer {
+    /// Constructor.
+    pub fn new_from_cam(logger:Logger, camera:&Camera2d) -> Self {
+        let this = Self::new(logger);
+        this.set_camera(camera);
+        this
+    }
```

Comment:
I would rename it to "new_with_cam", as this is not "conversion camera to layer".